tools/windows-node-installer/pkg/cloudprovider/azure/vm.go

```diff
@@ -0,0 +1,756 @@
+	ipAddress, ipErr := az.getIPAddress(ctx)
+	check(ipErr, fmt.Sprintf("couldn't get the IP Address for corresponding resource: %s", az.IpName))
+	if ipErr != nil {
+		ipAddress = az.IpName
+	}
+	resultPath := az.resourceTrackerDir + "/" + instanceName
+	f, err := os.Create(resultPath)
+	check(err, "unable to create a file")
+	defer f.Close()
+	_, err = f.Write([]byte(fmt.Sprintf("xfreerdp /u:windows /v:%s /h:1080 /w:1920 /p:%s \n", ipAddress, adminPassword)))
+	check(err, "unable to write data into file")
+	if err == nil {
+		log.Info(fmt.Sprintf("Please Check for file %s in %s directory on how to access the node",
+			instanceName, az.resourceTrackerDir))
+	} else {
+		log.Info(fmt.Sprintf("xfreerdp /u:windows /v:%s /h:1080 /w:1920 /p:%s", ipAddress, adminPassword))
+	}
+	return nil
+}
+
+// getNICname returns nicName by taking instance name as an argument.
+func (az *AzureProvider) getNICname(ctx context.Context, vmName string) (err error, nicName string) {
+	vmStruct, err := az.vmClient.Get(ctx, az.resourceGroupName, vmName, "instanceView")
+	if err != nil {
+		log.Error(err, fmt.Sprintf("cannot fetch the instance data of %s", vmName))
+		return
+	}
+	networkProfile := vmStruct.VirtualMachineProperties.NetworkProfile
+	networkInterface := (*networkProfile.NetworkInterfaces)[0]
+	nicID := *networkInterface.ID
+	nicName = extractResourceName(nicID)
+	return nil, nicName
+}
+
+// returnNicIpdetails returns ipName by taking instance name as an argument.
+func (az *AzureProvider) getIPname(ctx context.Context, vmName string) (err error, ipName string) {
+	vmStruct, err := az.vmClient.Get(ctx, az.resourceGroupName, vmName, "instanceView")
+	if err != nil {
+		log.Error(err, fmt.Sprintf("cannot fetch the instance data of %s", vmName))
+		return
+	}
+	networkProfile := vmStruct.VirtualMachineProperties.NetworkProfile
+	networkInterface := (*networkProfile.NetworkInterfaces)[0]
+	nicID := *networkInterface.ID
+	nicName := extractResourceName(nicID)
+	interfaceStruct, err := az.nicClient.Get(ctx, az.resourceGroupName, nicName, "")
+	if err != nil {
+		log.Error(err, fmt.Sprintf("cannot fetch the network interface data of %s", vmName))
+	}
+	interfacePropFormat := *(interfaceStruct.InterfacePropertiesFormat)
+	interfaceIPConfigs := *(interfacePropFormat.IPConfigurations)
+	ipConfigProp := *(interfaceIPConfigs[0].InterfaceIPConfigurationPropertiesFormat)
+	ipID := *(ipConfigProp.PublicIPAddress.ID)
+	ipName = extractResourceName(ipID)
+	return nil, ipName
+}
+
+// destroyIP deletes the IP address by taking it's name as argument.
+func (az *AzureProvider) destroyIP(ctx context.Context, ipName string) (err error) {
+	future, err := az.ipClient.Delete(ctx, az.resourceGroupName, ipName)
+	if err != nil {
+		log.Error(err, fmt.Sprintf("failed to delete the public IP: %s", ipName))
+		return
+	}
+	var response bool = false
+	var responseStatus int = 200
+	var r autorest.Response
+	for i := 1; i <= 300; i++ {
```

Comment:
Why 300? Are you planning to wait for 300? What's the reason behind this number?

Reply:
I came up through trial & error method. I haven't observed deletion of resource taking more than 5 mins.

Reply:
I am a nack on this. I again question why you need to wait for the deletion. Can't this be a fire and forget operation?

Reply:
No they can't be fire and forget operations, I need to get confirmation before I proceed to delete the next resource.

Reply:
I am not sure if all the resources are connected like the instance. What happens if you don't?